std/traits.d

```diff
@@ -5971,6 +5972,31 @@ enum bool isNarrowString(T) = (is(T : const char[]) || is(T : const wchar[])) &&
     }
 }
 
+/**
+ * Detects whether `T` is a comparable type. Basic types and structs and
+ * classes that implement opCmp are comparable.
+ */
+enum bool isComparable(T) = is(typeof({ return T.init < T.init; }));
```

Comment:
Given that it's only testing a single type, `hasOpCmp` would probably make a lot of sense, though if you want to be pedantic, built-in types don't technically have `opCmp`. "comparable" is certainly vague though given that both `opEquals` and `opCmp` have to do with comparisons.

There's also the question of whether this should really be taking two types, though that makes more sense with `opEquals` than with `opCmp`.

I don't think that `isComparable` is a bad name per se (particularly since coming up with a really good one isn't exactly easy), but I agree that it's more ambiguous than would be desirable.

Honestly, this is the sort of thing where I'd just make the test explicit in the template constraint rather than bothering with a helper template, but I see no problem with adding a helper template.